### ISSUE TITLE
CIMEROOT in your environment confuses scripts in CESM checkout

### ISSUE THREAD

**Issue by mnlevy1981** [ESMCI/cime#326]
(I assume this is pertinent to ACME checkouts as well, hence the issue here instead of [CESM-Development/cime](https://github.com/CESM-Development/cime))

I have `$CIMEROOT` set in my environment, pointing at my local clone of the stand-alone CIME repository. This confuses `create_newcase` and the other scripts, because it looks for the active components in `$CIMEROOT/../components` instead of `$PWD/../../components`; if I unset the environment variable then the scripts appear to fall back to the desired `$PWD/../../components` because everything works as expected.

I think I set this variable so that `machines/configure` would run correctly; @jedwards4b sent out an email in April 2015 where said

```
I've just updated the machines/configure file to the new cime directory structure and added a couple of new options/ features:

-cimeroot :  you can provide the cime root directory with a command line argument, if
                    not provided it will look for CIMEROOT in the environment if nether is present                     it will fail.

​-output_format:   Can be make or cmake, defaults to make which produces the original                                       Makefile Macros file - if set to cmake it will produce a Macros file 
                            suitable for use with cmake



We should make sure that all auxiliary tools in the cesm distribution know how to use a
configure generated Macros file to build.   If you need any help to do this with a particular tool please ask.
```

There are several mapping tools (`tools/mapping`) that depend on the `Macros` file to build correctly.

---

So basically there's a conflict between needing to have `$CIMEROOT` set when building some tools and needing to have `$CIMEROOT` NOT set when creating CESM (and potentially ACME) cases.

One possible fix would be to add using the `-cimeroot` option when calling `configure` to the mapping tool install directions, but that seems like a band-aid rather than a fix given that there are probably plenty of other tools that also call `configure`. Another would be to change the behavior of the scripts (to not look for `$CIMEROOT` from the environment) and / or `configure` (to not "require" having `$CIMEROOT` set). I'm not terribly familiar with either the scripts or `configure` so I don't know if those options are palatable, but maybe someone else can come up with a fix that works for everything?

**Comment by jgfouca** [comment]
Assigning @jedwards4b because I have too many issues already. Feel free to reassign.

**Comment by quantheory** [comment]
I will probably end up dealing with this when I finish moving `configure` to Python next week.

**Comment by douglasjacobsen** [comment]
This is more than just $CIMEROOT. After a discussion in slack, it seems like a user's environment variables override the values within the `env_*.xml`.

An example of this failing can be seen by doing the following:

```
./create_newcase -case test_compset -compset X -res f19_g16
cd test_compset
export PIO_VERSION=1.7.2
./case.setup
```

The result of which is the following error:
`ERROR: Entry PIO_REARRANGER was listed as type int but value '1.7.2' is not valid int`

Even though PIO_VERSION is set in env_build.xml with a value of `1`

**Comment by quantheory** [comment]
I think that @mnlevy1981's original complaint still stands, even though @douglasjacobsen's issue was addressed.

**Comment by douglasjacobsen** [comment]
@quantheory And it seems like I'm still having issues with `case.build` even though my issues with `case.setup` are fixed.

**Comment by douglasjacobsen** [comment]
@rljacob Why was this closed? I think the consensus was the the original issue hasn't been resolved yet, but the sub-issues, which are similar had been resolve.

**Comment by rljacob** [comment]
I misread the history. Re-opening

**Comment by rljacob** [comment]
Did issue #406 duplicate this one?

**Comment by mnlevy1981** [comment]
@rljacob I think so - I'll close this ticket and add a comment to #406 mentioning that CIMEROOT is another example of an environment variable that `create_newcase` is picking up
